src/ui/actions/comments.ts

```diff
@@ -58,13 +57,21 @@ export function createComment(
     const primaryLabel = labels?.primary;
     const secondaryLabel = labels?.secondary;
 
+    // If a comment is saved to Apollo, it will be assigned a persisted ID.
+    // Local, pending comments just need stable+unique IDs.
+    // Avoid using createdAt or updatedAt in these IDs though as these values are not stable.
+    const id = sourceLocation
```

Comment:
So, this is fine, but I just want to clear up some confusion around how `commentKey` works vs. what this change is doing. And a feelings disclaimer, I wrote that function and I feel like it's getting short-shrift here, and I know it's silly but I put a lot of work into it at the time and I feel a little defensive about it.

In the PR description it says:

> move away from using createdAt or updatedAt in keys. These values aren't stable and so they cause unnecessary unmounts and remounts within the React tree. (Even for comments that are persisted, Apollo will return a new createdAt value.)

It is true that after [this change](https://github.com/RecordReplay/devtools/commit/ab24d3f649689209e9ff3f2783ceb5b7ca878dbe#diff-9168052ad32419af17b20e0eaab3d17efed5744ba76105290e3d9c4b21d5295aR44) `commentKey`  no longer worked as intended, because it included an unstable value (`createdAt`) as you mention. However, previous to that change, this line, which generates an identifier for a comment by looking at its unique, stable identifiers, is *exactly* what `commentKey` was doing. Now, it's fine, we can delete all of that code and the accompanying tests, but I just want to point out that we already have all of this logic in a tested, (I think) well-named helpers file, and we are not moving away from that idea, but removing a regression introduced by the PR I linked above.

OK, sorry, rant over! I feel silly leaving this comment, but as I'm sure you can appreciate, it took up so much space in my brain when thinking about these problems, and I think we arrived at extremely similar solutions, which is great and gives me confidence in the direction that we are going (not because *I* was one of them, but because multiple people working independently made the same decisions).

Reply:
Ooops, I was replying to your comments in order and replied to the one above before seeing this one (so I deleted my outer reply and I'm going to respond here directly).

I appreciate the disclaimer ❤️  I'm sincerely sorry if my comments or code changes seemed inconsiderate or dismissive. That was not my intent.

When I stepped into this code, I was _really_ confused and it took me a lot of digging and notes-taking to wrap my brain around how it worked. A lot of the comments I left on this PR (and the related issues) were my best attempt and sorting through the mental space and organizing my thoughts around it.

You're right in pointing out that the new ID is the very similar to the previous one, minus the date. And yeah, if I look at an older commit, without the date, it's the same. So maybe I should find another way to (re)add tests for the spirit of what was there before.

Even though the code is doing a very similar thing, I think this is a small but nice change still because:
* Unique ID rather than a "magic" one.
* Setting `key={id}` rather than `key={someMappingFunction(object)}` is a more standard usage with React components.

Again though, I understand how this might have made you feel not great and I'm really sorry for that.

Reply:
Will re-add some comment ordering tests in #6567